tools/src/main/scala/scala/scalanative/linker/Reach.scala

```diff
@@ -41,18 +43,23 @@ class Reach(config: build.Config, entries: Seq[Global], loader: ClassLoader) {
 
   def cleanup(): Unit = {
     // Remove all unreachable methods from the
-    // responds map of every class. Optimizer and
-    // codegen may never increase reachability past
-    // what's known now, so it's safe to do this.
+    // responds and defaultResponds of every class.
+    // Optimizer and codegen may never increase reachability
+    // past what's known now, so it's safe to do this.
     infos.values.foreach {
       case cls: Class =>
-        val entries = cls.responds.toArray
-        entries.foreach {
+        cls.responds.foreach {
           case (sig, name) =>
             if (!done.contains(name)) {
               cls.responds -= sig
```

Comment:
It is not safe to mutate a collection while you're iterating on it. That was the purpose of the `.toArray` in the previous code. It should be kept.